<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html lang="en">
<head>
<meta http-equiv="Content-Type" content="text/html; charset=UTF-8">
<meta charset="utf-8">
<meta http-equiv="X-UA-Compatible" content="IE=edge">
<meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no">
<meta name="description" content="courseEpimutations">
<title>The epimutacions User's Guide • courseEpimutations</title>
<!-- favicons --><link rel="icon" type="image/png" sizes="16x16" href="../favicon-16x16.png">
<link rel="icon" type="image/png" sizes="32x32" href="../favicon-32x32.png">
<link rel="apple-touch-icon" type="image/png" sizes="180x180" href="../apple-touch-icon.png">
<link rel="apple-touch-icon" type="image/png" sizes="120x120" href="../apple-touch-icon-120x120.png">
<link rel="apple-touch-icon" type="image/png" sizes="76x76" href="../apple-touch-icon-76x76.png">
<link rel="apple-touch-icon" type="image/png" sizes="60x60" href="../apple-touch-icon-60x60.png">
<script src="../deps/jquery-3.6.0/jquery-3.6.0.min.js"></script><meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no">
<link href="../deps/bootstrap-5.1.3/bootstrap.min.css" rel="stylesheet">
<script src="../deps/bootstrap-5.1.3/bootstrap.bundle.min.js"></script><link href="../deps/_Source%20Sans%20Pro-0.4.1/font.css" rel="stylesheet">
<link href="../deps/_Source%20Code%20Pro-0.4.1/font.css" rel="stylesheet">
<!-- Font Awesome icons --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/all.min.css" integrity="sha256-mmgLkCYLUQbXn0B1SRqzHar6dCnv9oZFPEC1g1cwlkk=" crossorigin="anonymous">
<link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/v4-shims.min.css" integrity="sha256-wZjR52fzng1pJHwx4aV2AO3yyTOXrcDW7jBpJtTwVxw=" crossorigin="anonymous">
<!-- bootstrap-toc --><script src="https://cdn.rawgit.com/afeld/bootstrap-toc/v1.0.1/dist/bootstrap-toc.min.js"></script><!-- headroom.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/headroom.min.js" integrity="sha256-AsUX4SJE1+yuDu5+mAVzJbuYNPHj/WroHuZ8Ir/CkE0=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/jQuery.headroom.min.js" integrity="sha256-ZX/yNShbjqsohH1k95liqY9Gd8uOiE1S4vZc+9KQ1K4=" crossorigin="anonymous"></script><!-- clipboard.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/2.0.6/clipboard.min.js" integrity="sha256-inc5kl9MA1hkeYUt+EC3BhlIgyp/2jDIyBLS6k3UxPI=" crossorigin="anonymous"></script><!-- search --><script src="https://cdnjs.cloudflare.com/ajax/libs/fuse.js/6.4.6/fuse.js" integrity="sha512-zv6Ywkjyktsohkbp9bb45V6tEMoWhzFzXis+LrMehmJZZSys19Yxf1dopHx7WzIKxr5tK2dVcYmaCk2uqdjF4A==" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/autocomplete.js/0.38.0/autocomplete.jquery.min.js" integrity="sha512-GU9ayf+66Xx2TmpxqJpliWbT5PiGYxpaG8rfnBEk1LL8l1KGkRShhngwdXK1UgqhAzWpZHSiYPc09/NwDQIGyg==" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mark.js/8.11.1/mark.min.js" integrity="sha512-5CYOlHXGh6QpOFA/TeTylKLWfB3ftPsde7AnmhuitiTX4K5SqCLBeKro6sPS8ilsz1Q4NRx3v8Ko2IBiszzdww==" crossorigin="anonymous"></script><!-- pkgdown --><script src="../pkgdown.js"></script><meta property="og:title" content="The epimutacions User's Guide">
<meta property="og:description" content="courseEpimutations">
<meta property="og:image" content="/logo.png">
<!-- mathjax --><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/MathJax.js" integrity="sha256-nvJJv9wWKEm88qvoQl9ekL2J+k/RWIsaSScxxlsrv8k=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/config/TeX-AMS-MML_HTMLorMML.js" integrity="sha256-84DKXVJXs0/F8OTMzX4UR909+jtl4G7SPypPavF+GfA=" crossorigin="anonymous"></script><!--[if lt IE 9]>
<script src="https://oss.maxcdn.com/html5shiv/3.7.3/html5shiv.min.js"></script>
<script src="https://oss.maxcdn.com/respond/1.4.2/respond.min.js"></script>
<![endif]-->
</head>
<body>
    <a href="#main" class="visually-hidden-focusable">Skip to contents</a>
    

    <nav class="navbar fixed-top navbar-light navbar-expand-lg bg-none"><div class="container">
    
    <a class="navbar-brand me-2" href="../index.html">courseEpimutations</a>

    <small class="nav-text text-muted me-auto" data-bs-toggle="tooltip" data-bs-placement="bottom" title="">0.0.0.9000</small>

    
    <button class="navbar-toggler" type="button" data-bs-toggle="collapse" data-bs-target="#navbar" aria-controls="navbar" aria-expanded="false" aria-label="Toggle navigation">
      <span class="navbar-toggler-icon"></span>
    </button>

    <div id="navbar" class="collapse navbar-collapse ms-3">
      <ul class="navbar-nav me-auto">
<li class="nav-item">
  <a class="nav-link" href="../reference/index.html">Reference</a>
</li>
<li class="active nav-item dropdown">
  <a href="#" class="nav-link dropdown-toggle" data-bs-toggle="dropdown" role="button" aria-expanded="false" aria-haspopup="true" id="dropdown-articles">Articles</a>
  <div class="dropdown-menu" aria-labelledby="dropdown-articles">
    <a class="dropdown-item" href="../articles/epimutacions_cast.html">The epimutacions User's Guide</a>
  </div>
</li>
      </ul>
<form class="form-inline my-2 my-lg-0" role="search">
        <input type="search" class="form-control me-sm-2" aria-label="Toggle navigation" name="search-input" data-search-index="../search.json" id="search-input" placeholder="Search for" autocomplete="off">
</form>

      <ul class="navbar-nav"></ul>
</div>

    
  </div>
</nav><div class="container template-article">



<script src="epimutacions_cast_files/kePrint-0.0.1/kePrint.js"></script><link href="epimutacions_cast_files/lightable-0.0.1/lightable.css" rel="stylesheet">
<div class="row">
  <main id="main" class="col-md-9"><div class="page-header">
      <img src="../logo.png" class="logo" alt=""><h1>The epimutacions User's Guide</h1>
                        <h4 data-toc-skip class="author">Leire
Abarrategui</h4>
            <address class="author_afil">
      Bioinformatics Research Group in Epidemiology (BRGE)Barcelona
Insitute for Global Health (ISGlobal)Faculty of Medical Sciences,
Newcastle UniversityAutonomous University of Barcelona
(UAB)<br><a class="author_email" href="mailto:#"></a><a href="mailto:leire.abarrategui-martinez@newcastle.ac.uk" class="email">leire.abarrategui-martinez@newcastle.ac.uk</a>
      </address>
                              <h4 data-toc-skip class="author">Carles
Hernandez-Ferrer</h4>
            <address class="author_afil">
      Centro Nacional de Análisis Genómico (CNAG-CRG), Center for
Genomic RegulationBarcelona Institute of Science and Technology
(BIST)<br><a class="author_email" href="mailto:#"></a><a href="mailto:carles.hernandez@cnag.crg.eu" class="email">carles.hernandez@cnag.crg.eu</a>
      </address>
                              <h4 data-toc-skip class="author">Carlos
Ruiz-Arenas</h4>
            <address class="author_afil">
      Centro de Investigación Biomédica en Red de Enfermedades Raras
(CIBERER)Universitat Pompeu Fabra
(UPF)<br><a class="author_email" href="mailto:#"></a><a href="mailto:carlos.ruiza@upf.edu" class="email">carlos.ruiza@upf.edu</a>
      </address>
                              <h4 data-toc-skip class="author">Juan R.
Gonzalez</h4>
            <address class="author_afil">
      Bioinformatics Research Group in Epidemiology (BRGE),Barcelona
Insitute for Global Health (ISGlobal)Department of Mathematics,
Autonomous University of Barcelona
(UAB)<br><a class="author_email" href="mailto:#"></a><a href="mailto:juanr.gonzalez@isglobal.org" class="email">juanr.gonzalez@isglobal.org</a>
      </address>
                  
            <h4 data-toc-skip class="date">2022-11-12</h4>
      
      
      <div class="d-none name"><code>epimutacions_cast.Rmd</code></div>
    </div>

    
        <div class="abstract">
      <p class="abstract">Abstract</p>
      <p>Las epimutaciones son alteraciones raras en el patrón de
      metilación de un loci específico. Aunque se sabe que pueden ser la
      causa de algunas enfermedades genéticas, no existen métodos
      estandarizados para detectarlas y cuantificarlas. Esta vignette es
      una guía introductoria del paquete <code>epimutacions</code>. El
      paquete incluye varios métodos de detección de outliers para
      identificar epimutaciones en datos genómicos de metilación de ADN
      provenientes de microarray. Las áreas que cubre este documento
      son: (1) instalación del paquete; (2) carga y preprocesamiento de
      los datos; y (3) identificación, anotación y visualización de las
      epimutaciones.</p>
    </div>
    
<div class="section level2">
<h2 id="introducción">Introducción<a class="anchor" aria-label="anchor" href="#introducci%C3%B3n"></a>
</h2>
<div class="section level3">
<h3 id="background">Background<a class="anchor" aria-label="anchor" href="#background"></a>
</h3>
<p>Previamente al desarrollo de este paquete, existían dos enfoques
distintos a la hora de abordar la detección de epimutaciones en datos de
metilación de microarray: (1) primero identificar CpGs individualescon
valores outlier y después clusterizarlos en epimutaciones <span class="citation">(Barbosa et al. 2018)</span>; (2) primero definir
regiones candidatas con el algoritmo bumphunter y después testear la
significancia estadística con un MANOVA <span class="citation">(Aref-Eshghi et al. 2019)</span>. Aunque estos dos
métodos han sido definidos y usados, no están disponibles de manera
pública y nunca se han comparado sus rendimientos.</p>
<p>Por eso, hemos desarrollado el paquete de R/Bioconductor
<code>epimutacions</code>, que implementa estos dos enfoques (llamados
<code>quantile</code> y <code>manova</code> respectivamente) además de 4
métodos adicionales. El primero, llamado <code>beta</code>, sigue la
línea del primer enfoque pero usa una distribución específica a la hora
de detectar los outliers. Los otros 3, llamados <code>mlm</code>,
<code>mahdist</code> e <code>iForest</code>, siguen el segundo enfoque
pero aplican métodos distintos para evaluar la significancia estadística
de las regiones.</p>
<p>Además, el paquete <code>epimutacions</code> proporciona herramientas
para la normalización de los datos crudos de metilación de ADN de
microarray, así como para la visualización y anotación de las
epimutaciones.</p>
</div>
<div class="section level3">
<h3 id="metodología">Metodología<a class="anchor" aria-label="anchor" href="#metodolog%C3%ADa"></a>
</h3>
<p>EL paquete <code>epimutacions</code> lleva a cabo un análisis
genómico de metilación de ADN para identificar epimutaciones que podrían
ser consideradas como biomarcadores en aquellos pacientes que se
sospeche que pueden estar sufriendo alguna enfermedad genética.</p>
<p>Para estudios con un diseño caso-control, se debe usar la función
<code>epimutacions</code> que compara cada muestra caso contra un panel
de referencia (individuos sanos). Si por el contrario, sólo se dispone
de muestras caso, se debe usar la función
<code><a href="https://rdrr.io/pkg/epimutacions/man/epimutations_one_leave_out.html" class="external-link">epimutations_one_leave_out()</a></code>, que compara cada muestra caso
contra todas las demás, que se usan como control. Independientemente de
la función que usemos, podremos aplicar cualquiera de los métodos
anteriormente mencionados, que son: (1) Análisis Multivariado de
Varianza (<code>manova</code>), (2) Modelo Lineal Multivariante
(<code>mlm</code>), (3) Isolation Forest (<code>iForest</code>), (4)
Distancia robusta de Mahalanobis (<code>mahdist</code>) (5)
<code>quantile</code> y (6) <code>beta</code>.</p>
<p>Los métodos <code>quantile</code> y <code>beta</code> comparan los
valores de metilación de cada CpG con la referencia y después
clusterizan los significativos en epimutaciones. Para indentificar los
CpG outliers, <code>quantile</code> usa un sliding window, mientras que
<code>beta</code> aplica una distribución beta. Definimos una
epimutación como una ventana consecutiva de al menos 3 CpGs outliers con
una distancia máxima entre ellos de 1kb <span class="citation">(Barbosa
et al. 2018)</span>.</p>
<p>Por otro lado, <code>manova</code>, <code>mlm</code>,
<code>iForest</code> y <code>mahdist</code> definen primero las regiones
candidatas (regiones diferencialmente metiladas (DMRs)) con el algoritmo
bumphubter<span class="citation">(Jaffe et al. 2012)</span> y después
comprueban si en estas DMRs hay suficientes CpGs outliers.</p>
<div class="figure" style="text-align: center">
<img src="fig/implementation.png" alt="Implementation of each outlier detection method" width="90%"><p class="caption">
Implementation of each outlier detection method
</p>
</div>
</div>
</div>
<div class="section level2">
<h2 id="setup">Setup<a class="anchor" aria-label="anchor" href="#setup"></a>
</h2>
<div class="section level3">
<h3 id="instalación-de-los-paquetes">Instalación de los paquetes<a class="anchor" aria-label="anchor" href="#instalaci%C3%B3n-de-los-paquetes"></a>
</h3>
<p>Necesitamos instalar <code>epimutacions</code> y
<code>epimutacionsData</code></p>
<div class="sourceCode" id="cb1"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="kw">if</span> <span class="op">(</span><span class="op">!</span><span class="fu"><a href="https://rdrr.io/r/base/ns-load.html" class="external-link">requireNamespace</a></span><span class="op">(</span><span class="st">"BiocManager"</span>, quietly <span class="op">=</span> <span class="cn">TRUE</span><span class="op">)</span><span class="op">)</span></span>
<span>    <span class="fu"><a href="https://rdrr.io/r/utils/install.packages.html" class="external-link">install.packages</a></span><span class="op">(</span><span class="st">"BiocManager"</span><span class="op">)</span></span>
<span></span>
<span><span class="fu">BiocManager</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/pkg/BiocManager/man/install.html" class="external-link">install</a></span><span class="op">(</span><span class="st">"epimutacions"</span><span class="op">)</span></span></code></pre></div>
<div class="sourceCode" id="cb2"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu">BiocManager</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/pkg/BiocManager/man/install.html" class="external-link">install</a></span><span class="op">(</span><span class="st">"epimutacionsData"</span><span class="op">)</span></span></code></pre></div>
</div>
<div class="section level3">
<h3 id="carga-de-las-librerías">Carga de las librerías<a class="anchor" aria-label="anchor" href="#carga-de-las-librer%C3%ADas"></a>
</h3>
<div class="sourceCode" id="cb3"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va"><a href="https://github.com/isglobal-brge/epimutacions" class="external-link">epimutacions</a></span><span class="op">)</span></span></code></pre></div>
</div>
<div class="section level3">
<h3 id="inicio-rápido">Inicio Rápido<a class="anchor" aria-label="anchor" href="#inicio-r%C3%A1pido"></a>
</h3>
<p>La figura @ref(fig:workflow) explica los principales análisis del
paquete.</p>
<p>En líneas generales, podemos proporcionar dos tipos de input:</p>
<ul>
<li><ol style="list-style-type: decimal">
<li>Objeto <code>GenomicRatioSet</code> con todos los individuos (sólo
casos o casos + controles).</li>
</ol></li>
<li><ol start="2" style="list-style-type: decimal">
<li>Datos crudos de casos + Panel de referencia. Los casos como ficheros
<code>IDAT</code> y el panel de referencia, bien como un objeto
<code>RGChannelSet</code> o bien seleccionando alguno de los que el
paquete proporciona como ejemplo (section @ref(datasets)). Después
deberemos hacer un preprocesamiento con la función
<code><a href="https://rdrr.io/pkg/epimutacions/man/epi_preprocess.html" class="external-link">epi_preprocess()</a></code>para obtener un objeto
<code>GenomicRatioSet</code>.</li>
</ol></li>
</ul>
<p>El objeto <code>GenomicRatioSet</code> es el que se usa como input de
las funciones principales <code><a href="https://rdrr.io/pkg/epimutacions/man/epimutations.html" class="external-link">epimutations()</a></code> (para estudios
caso-control) o <code><a href="https://rdrr.io/pkg/epimutacions/man/epimutations_one_leave_out.html" class="external-link">epimutations_one_leave_out()</a></code> (para estudios
con sólo casos).</p>
<div class="figure" style="text-align: center">
<img src="fig/workflow.png" alt="Allowed data formats, normalization and input types" width="90%"><p class="caption">
Allowed data formats, normalization and input types
</p>
</div>
</div>
</div>
<div class="section level2">
<h2 id="datasets">Dataset<a class="anchor" aria-label="anchor" href="#datasets"></a>
</h2>
<p>El paquete contiene un dataset de prueba llamado <code>methy</code>.
Se trata de un objeto <code>GenomicRatioSet</code> que incluye 49
controles <a href="https://www.ncbi.nlm.nih.gov/geo/query/acc.cgi?acc=GSE104812" class="external-link">(GEO:
GSE104812)</a> y 3 casos <a href="https://www.ncbi.nlm.nih.gov/geo/query/acc.cgi" class="external-link">(GEO:
GSE97362)</a>.</p>
<p>Podemos acceder al dataset de la siguiente manera:</p>
<div class="sourceCode" id="cb4"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va">ExperimentHub</span><span class="op">)</span></span>
<span><span class="va">eh</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/ExperimentHub/man/ExperimentHub-class.html" class="external-link">ExperimentHub</a></span><span class="op">(</span><span class="op">)</span></span>
<span><span class="fu"><a href="https://rdrr.io/pkg/AnnotationHub/man/AnnotationHub-class.html" class="external-link">query</a></span><span class="op">(</span><span class="va">eh</span>, <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"epimutacionsData"</span><span class="op">)</span><span class="op">)</span></span></code></pre></div>
<pre><code>ExperimentHub with 3 records
# snapshotDate(): 2022-04-26
# $dataprovider: GEO, Illumina 450k array
# $species: Homo sapiens
# $rdataclass: RGChannelSet, GenomicRatioSet, GRanges
# additional mcols(): taxonomyid, genome, description,
#   coordinate_1_based, maintainer, rdatadateadded, preparerclass,
#   tags, rdatapath, sourceurl, sourcetype 
# retrieve records with, e.g., 'object[["EH6690"]]' 

           title                   
  EH6690 | Control and case samples
  EH6691 | Reference panel         
  EH6692 | Candidate epimutations  </code></pre>
<div class="sourceCode" id="cb6"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">methy</span> <span class="op">&lt;-</span> <span class="va">eh</span><span class="op">[[</span><span class="st">"EH6690"</span><span class="op">]</span><span class="op">]</span></span></code></pre></div>
<p>A parte, tenemos un <code>GenomicRatioSet</code> que sólo contiene
muestras de casos, lo podemos cargar así:</p>
<div class="sourceCode" id="cb7"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="https://rdrr.io/r/utils/data.html" class="external-link">data</a></span><span class="op">(</span><span class="va">GRset</span><span class="op">)</span></span></code></pre></div>
</div>
<div class="section level2">
<h2 id="preprocesamiento">Preprocesamiento<a class="anchor" aria-label="anchor" href="#preprocesamiento"></a>
</h2>
<p>Cómo nuestros datos ya están preprocesados, este paso no hará
falta.</p>
<p>Si tuviesemos datos crudos, la función <code><a href="https://rdrr.io/pkg/epimutacions/man/epi_preprocess.html" class="external-link">epi_preprocess()</a></code>
permite usar cualquiera de los 6 métodos de prepocesamiento del paquete
<code>minfi</code>. Para ello, debemos tener todos los ficheros
<code>IDAT</code> en un mismo directorio.a función se usa de la
siguiente manera:</p>
<div class="sourceCode" id="cb8"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">Dataset_procesado</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/epimutacions/man/epi_preprocess.html" class="external-link">epi_preprocess</a></span><span class="op">(</span><span class="va">directorio</span>, </span>
<span>                                  <span class="va">panel_referencia</span>, </span>
<span>                                  pattern <span class="op">=</span> <span class="st">"SampleSheet.csv"</span><span class="op">)</span></span></code></pre></div>
</div>
<div class="section level2">
<h2 id="epimutations">Epimutations<a class="anchor" aria-label="anchor" href="#epimutations"></a>
</h2>
<div class="section level3">
<h3 id="detección-de-epimutaciones">Detección de epimutaciones<a class="anchor" aria-label="anchor" href="#detecci%C3%B3n-de-epimutaciones"></a>
</h3>
<div class="section level4">
<h4 id="diseño-caso-control">Diseño caso-control<a class="anchor" aria-label="anchor" href="#dise%C3%B1o-caso-control"></a>
</h4>
<p>Primero usaremos el dataset <code>methy</code> (section @ref(methy))
que contiene muestras de casos y controles. Por lo tanto, usaremos la
función <code>epimutations</code>. Podemos escoger cualquiera de los 6
métodos implementados en el paquete: (1) Análisis Multivariado de
Varianza (<code>manova</code>), (2) Modelo Lineal Multivariante
(<code>mlm</code>), (3) Isolation Forest (<code>iForest</code>), (4)
Distancia robusta de Mahalanobis (<code>mahdist</code>) (5)
<code>quantile</code> y (6) <code>beta</code>.</p>
<p>Para empezar, cogemos el dataset y separamos los casos de los
controles:</p>
<div class="sourceCode" id="cb9"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">case_samples</span> <span class="op">&lt;-</span> <span class="va">methy</span><span class="op">[</span>,<span class="va">methy</span><span class="op">$</span><span class="va">status</span> <span class="op">==</span> <span class="st">"case"</span><span class="op">]</span></span>
<span><span class="va">control_samples</span> <span class="op">&lt;-</span> <span class="va">methy</span><span class="op">[</span>,<span class="va">methy</span><span class="op">$</span><span class="va">status</span> <span class="op">==</span> <span class="st">"control"</span><span class="op">]</span></span></code></pre></div>
<p>Usaremos el método <code>manova</code>. Si quisiéramos, podríamos
especificar el cromosoma o la región a analizar para reducir el tiempo
de ejecución:</p>
<div class="sourceCode" id="cb10"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">epi_mvo</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/epimutacions/man/epimutations.html" class="external-link">epimutations</a></span><span class="op">(</span><span class="va">case_samples</span>, </span>
<span>                        <span class="va">control_samples</span>, </span>
<span>                        method <span class="op">=</span> <span class="st">"manova"</span><span class="op">)</span></span></code></pre></div>
</div>
<div class="section level4">
<h4 id="diseño-sólo-casos">Diseño sólo casos<a class="anchor" aria-label="anchor" href="#dise%C3%B1o-s%C3%B3lo-casos"></a>
</h4>
<p>Para analizar el dataset que sólo contiene muestras de casos
<code>GRset</code>, usaremos la función
<code><a href="https://rdrr.io/pkg/epimutacions/man/epimutations_one_leave_out.html" class="external-link">epimutations_one_leave_out()</a></code>. En este caso no hace falta
dividir el dataset, simplemente corremos la función:</p>
<div class="sourceCode" id="cb11"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">epi_mva_one_leave_out</span><span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/epimutacions/man/epimutations_one_leave_out.html" class="external-link">epimutations_one_leave_out</a></span><span class="op">(</span><span class="va">GRset</span>,</span>
<span>                                                   method <span class="op">=</span> <span class="st">'manova'</span><span class="op">)</span></span></code></pre></div>
</div>
</div>
<div class="section level3">
<h3 id="parámetros">Parámetros<a class="anchor" aria-label="anchor" href="#par%C3%A1metros"></a>
</h3>
<p>Cada método de detección tiene sus propios parámetros. Los podemos
modificar mediante la función <code><a href="https://rdrr.io/pkg/epimutacions/man/epi_parameters.html" class="external-link">epi_parameters()</a></code>. Si la
corremos sin argumentos, nos muestra una lista de los parámetros por
defecto:</p>
<div class="sourceCode" id="cb12"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="https://rdrr.io/pkg/epimutacions/man/epi_parameters.html" class="external-link">epi_parameters</a></span><span class="op">(</span><span class="op">)</span></span></code></pre></div>
<pre><code>$manova
$manova$pvalue_cutoff
[1] 0.05


$mlm
$mlm$pvalue_cutoff
[1] 0.05


$iForest
$iForest$outlier_score_cutoff
[1] 0.7

$iForest$ntrees
[1] 100


$mahdist
$mahdist$nsamp
[1] "deterministic"


$quantile
$quantile$window_sz
[1] 1000

$quantile$offset_abs
[1] 0.15

$quantile$qsup
[1] 0.995

$quantile$qinf
[1] 0.005


$beta
$beta$pvalue_cutoff
[1] 1e-06

$beta$diff_threshold
[1] 0.1</code></pre>
<p>Los nombres de los parámetros modificables se encuentran en esta
tabla:</p>
<table class="table">
<thead><tr>
<th style="text-align:left;">
Parámetros
</th>
<th style="text-align:left;">
Descripción
</th>
</tr></thead>
<tbody>
<tr grouplength="1">
<td colspan="2" style="border-bottom: 1px solid;">
<strong>Manova, mlm</strong>
</td>
</tr>
<tr>
<td style="text-align:left;padding-left: 2em;" indentlevel="1">
pvalue_cutoff
</td>
<td style="text-align:left;">
El p-valor umbral para determinar qué regiones CpG son outliers
</td>
</tr>
<tr grouplength="2">
<td colspan="2" style="border-bottom: 1px solid;">
<strong>iso.forest</strong>
</td>
</tr>
<tr>
<td style="text-align:left;padding-left: 2em;" indentlevel="1">
outlier_score_cutoff
</td>
<td style="text-align:left;">
El umbral para determinar qué regiones CpG son outliers
</td>
</tr>
<tr>
<td style="text-align:left;padding-left: 2em;" indentlevel="1">
ntrees
</td>
<td style="text-align:left;">
El número de árboles binarios que se deben construir para el modelo
</td>
</tr>
<tr grouplength="1">
<td colspan="2" style="border-bottom: 1px solid;">
<strong>mahdist.mcd</strong>
</td>
</tr>
<tr>
<td style="text-align:left;padding-left: 2em;" indentlevel="1">
nsamp
</td>
<td style="text-align:left;">
El número de subgrupos usados para las estimaciones iniciales del MCD
</td>
</tr>
<tr grouplength="2">
<td colspan="2" style="border-bottom: 1px solid;">
<strong>quantile</strong>
</td>
</tr>
<tr>
<td style="text-align:left;padding-left: 2em;" indentlevel="1">
window_sz
</td>
<td style="text-align:left;">
La distancia máxima entre CpGs para que se consideren dentro de la misma
DMR
</td>
</tr>
<tr>
<td style="text-align:left;padding-left: 2em;" indentlevel="1">
offset_mean/offset_abs
</td>
<td style="text-align:left;">
El umbral superior e inferior para considerar un CpG como outlier
</td>
</tr>
<tr grouplength="2">
<td colspan="2" style="border-bottom: 1px solid;">
<strong>beta</strong>
</td>
</tr>
<tr>
<td style="text-align:left;padding-left: 2em;" indentlevel="1">
pvalue_cutoff
</td>
<td style="text-align:left;">
El p-valor mínimo para considerar un CpG como outlier
</td>
</tr>
<tr>
<td style="text-align:left;padding-left: 2em;" indentlevel="1">
diff_threshold
</td>
<td style="text-align:left;">
La diferencia de metilación mínima entre el CpG y la media para
considerar una posición como outlier
</td>
</tr>
</tbody>
</table>
<p>Si quisiéramos cambiar algun parámetro, se puede hacer utilizando los
argumentos que vemos en la tabla. Por ejemplo, para modificar el p-valor
de corte de manova:</p>
<div class="sourceCode" id="cb14"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">parameters</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/epimutacions/man/epi_parameters.html" class="external-link">epi_parameters</a></span><span class="op">(</span>manova <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/list.html" class="external-link">list</a></span><span class="op">(</span><span class="st">"pvalue_cutoff"</span> <span class="op">=</span> <span class="fl">0.01</span><span class="op">)</span><span class="op">)</span></span>
<span><span class="va">parameters</span><span class="op">$</span><span class="va">manova</span><span class="op">$</span><span class="va">pvalue_cutoff</span></span></code></pre></div>
<pre><code>[1] 0.01</code></pre>
</div>
<div class="section level3">
<h3 id="descripción-de-los-resultados">Descripción de los resultados:<a class="anchor" aria-label="anchor" href="#descripci%C3%B3n-de-los-resultados"></a>
</h3>
<p>Las funciones <code>epimutations</code> y
<code><a href="https://rdrr.io/pkg/epimutacions/man/epimutations_one_leave_out.html" class="external-link">epimutations_one_leave_out()</a></code> devuelven un data frame con
todas las epimutaciones que se hayan detectado y los siguientes
argumentos:</p>
<table class="table">
<colgroup>
<col width="16%">
<col width="83%">
</colgroup>
<thead><tr class="header">
<th>Column name</th>
<th>Description</th>
</tr></thead>
<tbody>
<tr class="odd">
<td><code>epi_id</code></td>
<td>Nombre sistemático de cada una de las epimutaciones detectadas</td>
</tr>
<tr class="even">
<td><code>sample</code></td>
<td>Nombre de la muestra en la que se ha encontrado la epimutación</td>
</tr>
<tr class="odd">
<td>
<code>chromosome</code> <code>start</code> <code>end</code>
</td>
<td>La localización de la epimutación</td>
</tr>
<tr class="even">
<td><code>sz</code></td>
<td>La medida de la ventana</td>
</tr>
<tr class="odd">
<td><code>cpg_ids</code></td>
<td>El número de CpGs en la epimutación</td>
</tr>
<tr class="even">
<td><code>cpg_n</code></td>
<td>Los nombres de los CpGs de la epimutación</td>
</tr>
<tr class="odd">
<td><code>outlier_score</code></td>
<td>En el método <code>manova</code> proporciona la aproximación al
F-test y el score de Pillai, separados por <code>/</code><br> En el
método <code>mlm</code> proporciona la aproximación al F-test y la R2
del modelo, separados por <code>/</code><br> En el método
<code>iForest</code> proporciona la magnitud del score del outlier.<br>
En el método <code>beta</code> proporciona la media de p-valores de los
CpGs de la DMR en cuestión.<br> En los métodos <code>quantile</code> y
<code>mahdist</code> se rellena con <code>NA</code>.</td>
</tr>
<tr class="even">
<td><code>pvalue</code></td>
<td>En los métodos <code>manova</code> y <code>mlm</code> proporciona el
p-valor obtenido del modeo.<br> En los métodos <code>quantile</code>,
<code>iForest</code>, <code>beta</code> y <code>mahdist</code> se
rellena con <code>NA</code>.</td>
</tr>
<tr class="odd">
<td><code>outlier_direction</code></td>
<td>Indica la dirección del outlier con “hypomethylation” y
“hypermethylation”.<br> En <code>manova</code>, <code>mlm</code>,
<code>iForest</code>, y <code>mahdist</code> se calcula a partir de los
valores obtenidos de <code>bumphunter</code>.<br> En <code>beta</code>
se calcula con el p-valor de cada CpG usando <code>diff_threshold</code>
y <code>pvalue_threshold</code>.<br> En <code>quantile</code> se calcula
a partir de la localización de la muestra en la distribución de la
referencia (outlier por la derecha vs. outlier por la izquierda).</td>
</tr>
<tr class="even">
<td><code>adj_pvalue</code></td>
<td>En los métodos <code>manova</code> y <code>mlm</code> proporcinoa el
p-valor ajustado con Benjamini-Hochberg basado en el número total de
regiones detectadas por <code>bumphunter</code>.<br> En los métodos
<code>quantile</code>, <code>iForest</code>, <code>mahdist</code> y
<code>beta</code> se rellena con <code>NA</code>.</td>
</tr>
<tr class="odd">
<td><code>epi_region_id</code></td>
<td>Nombre de la epimutación tal y como esté definido en
<code>candRegsGR</code>.</td>
</tr>
<tr class="even">
<td><code>CRE</code></td>
<td>cREs (cis-Regulatory Elements) definidos por ENCODE que se
superponen con la región de la epimutación.</td>
</tr>
<tr class="odd">
<td><code>CRE_type</code></td>
<td>Tipos de cREs (cis-Regulatory Elements) definidos por ENCODE.</td>
</tr>
</tbody>
</table>
<p>A modo de ejemplo, vamos a visualizar los resultados que hemos
obtenido con el método ANOVA y la función <code>epimutations</code>
(diseño caso-control) - (<code>epi_mvo</code>):</p>
<div class="sourceCode" id="cb16"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="https://rdrr.io/r/base/dim.html" class="external-link">dim</a></span><span class="op">(</span><span class="va">epi_mvo</span><span class="op">)</span></span></code></pre></div>
<pre><code>[1] 51 16</code></pre>
<div class="sourceCode" id="cb18"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="https://rdrr.io/r/base/class.html" class="external-link">class</a></span><span class="op">(</span><span class="va">epi_mvo</span><span class="op">)</span></span></code></pre></div>
<pre><code>[1] "tbl_df"     "tbl"        "data.frame"</code></pre>
<div class="sourceCode" id="cb20"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="https://rdrr.io/r/utils/head.html" class="external-link">head</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/pkg/BiocGenerics/man/as.data.frame.html" class="external-link">as.data.frame</a></span><span class="op">(</span><span class="va">epi_mvo</span><span class="op">)</span>, <span class="fl">12</span><span class="op">)</span></span></code></pre></div>
<pre><code>          epi_id     sample chromosome     start       end  sz cpg_n
1   epi_manova_1 GSM2562699      chr19  12777736  12777903 167     7
2   epi_manova_2 GSM2562699       chr7  90895894  90896701 807     4
3   epi_manova_3 GSM2562699       chr7  27186993  27187691 698    11
4   epi_manova_4 GSM2562699      chr19  12758416  12759156 740     6
5   epi_manova_5 GSM2562699       chr7     31006     31497 491     6
6   epi_manova_6 GSM2562699       chr7   5271480   5271655 175     4
7   epi_manova_7 GSM2562699       chr7   1080558   1081068 510     4
8   epi_manova_8 GSM2562699       chr7   1209495   1209742 247     3
9   epi_manova_9 GSM2562699       chr7   1003645   1003750 105     3
10 epi_manova_10 GSM2562699       chr8 144810034 144810657 623     3
11 epi_manova_11 GSM2562699      chr19  44278274  44278628 354     3
12 epi_manova_12 GSM2562699      chr19  44285297  44285568 271     3
                                                                                                                    cpg_ids
1                                              cg20791841,cg25267526,cg03641858,cg25441478,cg14132016,cg23954461,cg03143365
2                                                                               cg08714590,cg01480180,cg21104276,cg17497608
3  cg23129930,cg05928186,cg06237983,cg04265576,cg12810523,cg03529432,cg22469274,cg17994139,cg14044640,cg09936824,cg12110087
4                                                         cg21535253,cg17737967,cg18339718,cg00422699,cg12063490,cg18530645
5                                                         cg00725145,cg16080333,cg23568068,cg05107246,cg15535638,cg20044143
6                                                                               cg26660312,cg00579868,cg09203501,cg00393985
7                                                                               cg03916490,cg12581298,cg22785556,cg24352561
8                                                                                          cg03188948,cg18402987,cg23708337
9                                                                                          cg08735705,cg12575883,cg07930192
10                                                                                         cg25468618,cg01399317,cg13735469
11                                                                                         cg03731131,cg11624345,cg22904711
12                                                                                         cg15977816,cg01424145,cg14066757
                        outlier_score outlier_direction       pvalue
1  302.558383959434/0.981008923520811  hypermethylation 3.441085e-33
2  27.7048116720887/0.715797609527385   hypomethylation 1.598679e-11
3  7.56010313828286/0.692080137662987  hypermethylation 1.297922e-06
4   5.8752838326799/0.456322665118055  hypermethylation 1.624372e-04
5  6.59037882739006/0.484929736771414   hypomethylation 5.827320e-05
6  7.33624119121713/0.400095151166048  hypermethylation 1.285674e-04
7  816.680878011864/0.986709853649848   hypomethylation 1.185295e-40
8   17.011266584818/0.531414979777337  hypermethylation 1.581504e-07
9  9.18210526711306/0.379706612211322  hypermethylation 7.468475e-05
10 8.02826408750705/0.348626542452343   hypomethylation 2.159693e-04
11 22.1696264146171/0.596444692968419   hypomethylation 5.786402e-09
12   12.7571508182161/0.4595987139229  hypermethylation 3.659987e-06
     adj_pvalue delta_beta  epi_region_id
1  2.236705e-31  0.2960902 chr19_12776725
2  1.023154e-09  0.4637789  chr7_90892836
3  7.917322e-05  0.1546873  chr7_27186554
4  9.096482e-03  0.2333481 chr19_12758416
5  3.438119e-03  0.1965240     chr7_31006
6  7.328345e-03  0.2721079   chr7_5270512
7  7.822945e-39  0.2639854   chr7_1080473
8  9.805325e-06  0.3103321   chr7_1209495
9  4.331715e-03  0.2275538   chr7_1003645
10 1.187831e-02  0.1879683 chr8_144807941
11 3.645433e-07  0.1870679 chr19_44277951
12 2.195992e-04  0.1488268 chr19_44285297
                                                                                                                                              CRE
1                                                                                                          EH38E1939817,EH38E1939818,EH38E1939819
2  EH38E2570884,EH38E2570885,EH38E2570886,EH38E2570887,EH38E2570888,EH38E2570889,EH38E2570890,EH38E2570891,EH38E2570892,EH38E2570893,EH38E2570894
3                                         EH38E2542602,EH38E2542603,EH38E2542604,EH38E2542605,EH38E2542606,EH38E2542608,EH38E2542601,EH38E2542607
4                                                                                                          EH38E1939806,EH38E1939807,EH38E1939808
5                                                                                                                                                
6                                                                                EH38E2531316,EH38E2531317,EH38E2531318,EH38E2531319,EH38E2531320
7                                                                                                          EH38E2528180,EH38E2528182,EH38E2528181
8                                                                                                                                    EH38E2528328
9                                                                   EH38E2528080,EH38E2528081,EH38E2528082,EH38E2528083,EH38E2528084,EH38E2528085
10                                                                                                         EH38E2675494,EH38E2675495,EH38E2675489
11                                                                                                                      EH38E1956415,EH38E1956416
12                                                                                                                      EH38E1956428,EH38E1956429
                                                                                                                                                                         CRE_type
1                                                                                                                                             pELS;pELS,CTCF-bound;PLS,CTCF-bound
2  pELS,CTCF-bound;pELS,CTCF-bound;pELS,CTCF-bound;PLS,CTCF-bound;pELS,CTCF-bound;pELS,CTCF-bound;pELS,CTCF-bound;pELS,CTCF-bound;dELS,CTCF-bound;dELS,CTCF-bound;dELS,CTCF-bound
3                                                                                                           PLS,CTCF-bound;pELS;PLS;pELS,CTCF-bound;pELS,CTCF-bound;pELS;PLS;pELS
4                                                                                                                                  pELS,CTCF-bound;pELS,CTCF-bound;PLS,CTCF-bound
5                                                                                                                                                                                
6                                                                                                            dELS;dELS,CTCF-bound;dELS,CTCF-bound;dELS,CTCF-bound;dELS,CTCF-bound
7                                                                                                                                            dELS,CTCF-bound;dELS;dELS,CTCF-bound
8                                                                                                                                                                 dELS,CTCF-bound
9                                                                                            dELS;dELS,CTCF-bound;dELS,CTCF-bound;dELS,CTCF-bound;dELS,CTCF-bound;dELS,CTCF-bound
10                                                                                                                                           dELS;dELS,CTCF-bound;dELS,CTCF-bound
11                                                                                                                                                           dELS;dELS,CTCF-bound
12                                                                                                                                                            PLS,CTCF-bound;pELS</code></pre>
<div class="sourceCode" id="cb22"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">epi_mvo</span><span class="op">[</span><span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="fl">10</span><span class="op">:</span><span class="fl">15</span><span class="op">)</span>,<span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="fl">12</span><span class="op">)</span><span class="op">]</span> <span class="op">&lt;-</span> <span class="va">epi_mvo</span><span class="op">[</span><span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="fl">10</span><span class="op">:</span><span class="fl">15</span><span class="op">)</span>,<span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="fl">12</span><span class="op">)</span><span class="op">]</span><span class="op">*</span><span class="fl">10</span></span>
<span><span class="fu"><a href="https://rdrr.io/pkg/BiocGenerics/man/as.data.frame.html" class="external-link">as.data.frame</a></span><span class="op">(</span><span class="va">epi_mvo</span><span class="op">)</span><span class="op">[</span><span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="fl">10</span><span class="op">:</span><span class="fl">15</span><span class="op">)</span>,<span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="fl">1</span>,<span class="fl">7</span>,<span class="fl">11</span>,<span class="fl">12</span><span class="op">)</span><span class="op">]</span></span></code></pre></div>
<pre><code>          epi_id cpg_n       pvalue   adj_pvalue
10 epi_manova_10     3 2.159693e-04 1.187831e-01
11 epi_manova_11     3 5.786402e-09 3.645433e-06
12 epi_manova_12     3 3.659987e-06 2.195992e-03
13 epi_manova_13     3 8.670808e-04 4.682236e-01
14 epi_manova_14    11 1.383286e-07 1.037465e-04
15 epi_manova_15     5 2.107632e-05 1.454266e-02</code></pre>
<div class="sourceCode" id="cb24"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu">kableExtra</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/pkg/knitr/man/kable.html" class="external-link">kable</a></span><span class="op">(</span></span>
<span>    <span class="va">epi_mvo</span><span class="op">[</span> <span class="fl">9</span>, <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="fl">1</span>,<span class="fl">2</span>, <span class="fl">7</span>, <span class="fl">10</span>,<span class="fl">11</span>,<span class="fl">12</span><span class="op">)</span><span class="op">]</span>,</span>
<span>    row.names <span class="op">=</span> <span class="cn">FALSE</span> <span class="op">)</span> <span class="op">%&gt;%</span>  <span class="fu">column_spec</span><span class="op">(</span><span class="fl">1</span><span class="op">:</span><span class="fl">6</span>, width <span class="op">=</span> <span class="st">"4cm"</span><span class="op">)</span></span></code></pre></div>
<table class="table">
<thead><tr>
<th style="text-align:left;">
epi_id
</th>
<th style="text-align:left;">
sample
</th>
<th style="text-align:right;">
cpg_n
</th>
<th style="text-align:left;">
outlier_direction
</th>
<th style="text-align:right;">
pvalue
</th>
<th style="text-align:right;">
adj_pvalue
</th>
</tr></thead>
<tbody><tr>
<td style="text-align:left;width: 4cm; ">
epi_manova_9
</td>
<td style="text-align:left;width: 4cm; ">
GSM2562699
</td>
<td style="text-align:right;width: 4cm; ">
3
</td>
<td style="text-align:left;width: 4cm; ">
hypermethylation
</td>
<td style="text-align:right;width: 4cm; ">
7.47e-05
</td>
<td style="text-align:right;width: 4cm; ">
0.0043317
</td>
</tr></tbody>
</table>
<div class="sourceCode" id="cb25"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">epi_mvo</span><span class="op">[</span><span class="va">epi_mvo</span><span class="op">$</span><span class="va">adj_pvalue</span><span class="op">&gt;</span><span class="fl">0.02</span>,<span class="op">]</span></span></code></pre></div>
<pre><code> [38;5;246m# A tibble: 6 × 16 [39m
  epi_id  sample chrom…¹  start    end    sz cpg_n cpg_ids outli…² outli…³
   [3m [38;5;246m&lt;chr&gt; [39m [23m    [3m [38;5;246m&lt;chr&gt; [39m [23m   [3m [38;5;246m&lt;chr&gt; [39m [23m     [3m [38;5;246m&lt;int&gt; [39m [23m   [3m [38;5;246m&lt;int&gt; [39m [23m  [3m [38;5;246m&lt;int&gt; [39m [23m  [3m [38;5;246m&lt;dbl&gt; [39m [23m  [3m [38;5;246m&lt;chr&gt; [39m [23m    [3m [38;5;246m&lt;chr&gt; [39m [23m    [3m [38;5;246m&lt;chr&gt; [39m [23m  
 [38;5;250m1 [39m epi_ma… GSM25… chr8    1.45 [38;5;246me [39m8 1.45 [38;5;246me [39m8   623     3 cg2546… 8.0282… hypome…
 [38;5;250m2 [39m epi_ma… GSM25… chr19   8.27 [38;5;246me [39m6 8.27 [38;5;246me [39m6   188     3 cg0668… 6.5923… hyperm…
 [38;5;250m3 [39m epi_ma… GSM25… chr7    1.55 [38;5;246me [39m8 1.55 [38;5;246me [39m8   746     5 cg1193… 5.8710… hyperm…
 [38;5;250m4 [39m epi_ma… GSM25… chr7    1.40 [38;5;246me [39m8 1.40 [38;5;246me [39m8   312     4 cg0824… 6.0014… hypome…
 [38;5;250m5 [39m epi_ma… GSM25… chr21   1.51 [38;5;246me [39m7 1.51 [38;5;246me [39m7   731     3 cg1229… 7.2652… hypome…
 [38;5;250m6 [39m epi_ma… GSM25… chr19   3.89 [38;5;246me [39m7 3.89 [38;5;246me [39m7   523     3 cg0275… 7.5347… hypome…
 [38;5;246m# … with 6 more variables: pvalue &lt;dbl&gt;, adj_pvalue &lt;dbl&gt;, [39m
 [38;5;246m#   delta_beta &lt;dbl&gt;, epi_region_id &lt;chr&gt;, CRE &lt;chr&gt;, CRE_type &lt;chr&gt;, and [39m
 [38;5;246m#   abbreviated variable names ¹​chromosome, ²​outlier_score, [39m
 [38;5;246m#   ³​outlier_direction [39m</code></pre>
</div>
<div class="section level3">
<h3 id="anotación-de-las-epimutaciones">Anotación de las epimutaciones<a class="anchor" aria-label="anchor" href="#anotaci%C3%B3n-de-las-epimutaciones"></a>
</h3>
<p>La función <code><a href="https://rdrr.io/pkg/epimutacions/man/annotate_epimutations.html" class="external-link">annotate_epimutations()</a></code> lleva a cabo un
enriquecimiento de las epimutaciones que se hayan detectado. Aporta
información sobre:</p>
<div class="sourceCode" id="cb27"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">rst_mvo</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/epimutacions/man/annotate_epimutations.html" class="external-link">annotate_epimutations</a></span><span class="op">(</span><span class="va">epi_mvo</span>, omim <span class="op">=</span> <span class="cn">FALSE</span><span class="op">)</span></span></code></pre></div>
<table class="table">
<colgroup>
<col width="23%">
<col width="76%">
</colgroup>
<thead><tr class="header">
<th>Column name</th>
<th>Description</th>
</tr></thead>
<tbody>
<tr class="odd">
<td><code>GencodeBasicV12_NAME</code></td>
<td>Nombres de los genes en GENECODE</td>
</tr>
<tr class="even">
<td><code>Regulatory_Feature_Group</code></td>
<td>Descripción del Methylation Consortium sobre la región
reguladora</td>
</tr>
<tr class="odd">
<td><code>Relation_to_Island</code></td>
<td>Localización del CpG en relación a la isla de CpGs</td>
</tr>
<tr class="even">
<td>
<code>OMIM_ACC</code><br><code>OMIM_DESC</code>
</td>
<td>Número de acceso y descripción de OMIM</td>
</tr>
<tr class="odd">
<td>
<code>ensembl_reg_id</code><br><code>ensembl_reg_coordinates</code><br><code>ensembl_reg_type</code><br><code>ensembl_reg_tissues</code>
</td>
<td>ID de la región, coordenadas, tipo y tejido, de Ensembl</td>
</tr>
</tbody>
</table>
<p>Podemos visualizar algunas anotacinoes de la siguiente manera:</p>
<div class="sourceCode" id="cb28"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu">kableExtra</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/pkg/knitr/man/kable.html" class="external-link">kable</a></span><span class="op">(</span></span>
<span>    <span class="va">rst_mvo</span><span class="op">[</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="fl">27</span>,<span class="fl">32</span><span class="op">)</span> ,<span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"epi_id"</span>, <span class="st">"cpg_ids"</span>, <span class="st">"Relation_to_Island"</span><span class="op">)</span><span class="op">]</span>,</span>
<span>    row.names <span class="op">=</span> <span class="cn">FALSE</span><span class="op">)</span> <span class="op">%&gt;%</span> <span class="fu">column_spec</span><span class="op">(</span><span class="fl">1</span><span class="op">:</span><span class="fl">3</span>, width <span class="op">=</span> <span class="st">"4cm"</span><span class="op">)</span></span></code></pre></div>
<table class="table">
<thead><tr>
<th style="text-align:left;">
epi_id
</th>
<th style="text-align:left;">
cpg_ids
</th>
<th style="text-align:left;">
Relation_to_Island
</th>
</tr></thead>
<tbody>
<tr>
<td style="text-align:left;width: 4cm; ">
epi_manova_44
</td>
<td style="text-align:left;width: 4cm; ">
cg19560927,cg02876326,cg16167052
</td>
<td style="text-align:left;width: 4cm; ">
N_Shore///N_Shore///Island
</td>
</tr>
<tr>
<td style="text-align:left;width: 4cm; ">
epi_manova_51
</td>
<td style="text-align:left;width: 4cm; ">
cg01396855,cg08684893,cg02386644
</td>
<td style="text-align:left;width: 4cm; ">
N_Shore///N_Shore///N_Shore
</td>
</tr>
</tbody>
</table>
<div class="sourceCode" id="cb29"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu">kableExtra</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/pkg/knitr/man/kable.html" class="external-link">kable</a></span><span class="op">(</span></span>
<span>    <span class="va">rst_mvo</span><span class="op">[</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="fl">4</span>,<span class="fl">8</span>,<span class="fl">22</span><span class="op">)</span> , <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"epi_id"</span>, <span class="st">"OMIM_ACC"</span>, <span class="st">"OMIM_DESC"</span><span class="op">)</span><span class="op">]</span>,</span>
<span>    row.names <span class="op">=</span> <span class="cn">FALSE</span> <span class="op">)</span> <span class="op">%&gt;%</span> <span class="fu">column_spec</span><span class="op">(</span><span class="fl">1</span><span class="op">:</span><span class="fl">3</span>, width <span class="op">=</span> <span class="st">"4cm"</span><span class="op">)</span></span></code></pre></div>
<div class="sourceCode" id="cb30"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu">kableExtra</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/pkg/knitr/man/kable.html" class="external-link">kable</a></span><span class="op">(</span></span>
<span>    <span class="va">rst_mvo</span><span class="op">[</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="fl">1</span><span class="op">:</span><span class="fl">5</span><span class="op">)</span>, <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"epi_id"</span>, <span class="st">"ensembl_reg_id"</span>, <span class="st">"ensembl_reg_type"</span><span class="op">)</span><span class="op">]</span>,</span>
<span>    row.names <span class="op">=</span> <span class="cn">FALSE</span> <span class="op">)</span> <span class="op">%&gt;%</span>  <span class="fu">column_spec</span><span class="op">(</span><span class="fl">1</span><span class="op">:</span><span class="fl">3</span>, width <span class="op">=</span> <span class="st">"4cm"</span><span class="op">)</span></span></code></pre></div>
<table class="table">
<thead><tr>
<th style="text-align:left;">
epi_id
</th>
<th style="text-align:left;">
ensembl_reg_id
</th>
<th style="text-align:left;">
ensembl_reg_type
</th>
</tr></thead>
<tbody>
<tr>
<td style="text-align:left;width: 4cm; ">
epi_manova_1
</td>
<td style="text-align:left;width: 4cm; ">
ENSR00000287480
</td>
<td style="text-align:left;width: 4cm; ">
Promoter
</td>
</tr>
<tr>
<td style="text-align:left;width: 4cm; ">
epi_manova_2
</td>
<td style="text-align:left;width: 4cm; ">
ENSR00000215007
</td>
<td style="text-align:left;width: 4cm; ">
Promoter
</td>
</tr>
<tr>
<td style="text-align:left;width: 4cm; ">
epi_manova_3
</td>
<td style="text-align:left;width: 4cm; ">
ENSR00000325316
</td>
<td style="text-align:left;width: 4cm; ">
Promoter
</td>
</tr>
<tr>
<td style="text-align:left;width: 4cm; ">
epi_manova_4
</td>
<td style="text-align:left;width: 4cm; ">
ENSR00001156259///ENSR00001156260
</td>
<td style="text-align:left;width: 4cm; ">
Promoter///CTCF Binding Site
</td>
</tr>
<tr>
<td style="text-align:left;width: 4cm; ">
epi_manova_5
</td>
<td style="text-align:left;width: 4cm; ">
ENSR00000324412
</td>
<td style="text-align:left;width: 4cm; ">
CTCF Binding Site
</td>
</tr>
</tbody>
</table>
<div class="sourceCode" id="cb31"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu">kableExtra</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/pkg/knitr/man/kable.html" class="external-link">kable</a></span><span class="op">(</span></span>
<span>    <span class="va">rst_mvo</span><span class="op">[</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="fl">1</span><span class="op">:</span><span class="fl">5</span><span class="op">)</span>, <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"GencodeBasicV12_NAME"</span>, <span class="st">"epi_id"</span>, <span class="st">"ensembl_reg_id"</span>, <span class="st">"ensembl_reg_type"</span><span class="op">)</span><span class="op">]</span>,</span>
<span>    row.names <span class="op">=</span> <span class="cn">FALSE</span> <span class="op">)</span> <span class="op">%&gt;%</span>  <span class="fu">column_spec</span><span class="op">(</span><span class="fl">1</span><span class="op">:</span><span class="fl">3</span>, width <span class="op">=</span> <span class="st">"4cm"</span><span class="op">)</span></span></code></pre></div>
<table class="table">
<thead><tr>
<th style="text-align:left;">
GencodeBasicV12_NAME
</th>
<th style="text-align:left;">
epi_id
</th>
<th style="text-align:left;">
ensembl_reg_id
</th>
<th style="text-align:left;">
ensembl_reg_type
</th>
</tr></thead>
<tbody>
<tr>
<td style="text-align:left;width: 4cm; ">
MAN2B1;MAN2B1;WDR83;WDR83 /// MAN2B1;MAN2B1;WDR83;WDR83 ///
WDR83;MAN2B1;MAN2B1;WDR83 /// WDR83;MAN2B1;MAN2B1;WDR83 ///
WDR83;MAN2B1;MAN2B1;WDR83 /// MAN2B1;MAN2B1;WDR83 ///
MAN2B1;MAN2B1;WDR83
</td>
<td style="text-align:left;width: 4cm; ">
epi_manova_1
</td>
<td style="text-align:left;width: 4cm; ">
ENSR00000287480
</td>
<td style="text-align:left;">
Promoter
</td>
</tr>
<tr>
<td style="text-align:left;width: 4cm; ">
FZD1;FZD1 /// FZD1;FZD1 /// FZD1;FZD1
</td>
<td style="text-align:left;width: 4cm; ">
epi_manova_2
</td>
<td style="text-align:left;width: 4cm; ">
ENSR00000215007
</td>
<td style="text-align:left;">
Promoter
</td>
</tr>
<tr>
<td style="text-align:left;width: 4cm; ">
HOXA6;HOXA-AS3;HOXA-AS3;HOXA-AS3;HOXA6;HOXA3;HOXA-AS3;HOXA-AS3 ///
HOXA6;HOXA-AS3;HOXA6;HOXA-AS3;HOXA-AS3;HOXA3;HOXA-AS3;HOXA-AS3 ///
HOXA6;HOXA-AS3;HOXA6;HOXA-AS3;HOXA6;HOXA3;HOXA-AS3;HOXA-AS3 ///
HOXA6;HOXA-AS3;HOXA6;HOXA-AS3;HOXA6;HOXA3;HOXA-AS3;HOXA-AS3 ///
HOXA6;HOXA-AS3;HOXA-AS3;HOXA6;HOXA3;HOXA-AS3;HOXA-AS3 ///
HOXA6;HOXA-AS3;HOXA-AS3;HOXA6;HOXA3;HOXA-AS3;HOXA-AS3 ///
HOXA6;HOXA-AS3;HOXA-AS3;HOXA6;HOXA3;HOXA-AS3;HOXA-AS3 ///
HOXA6;HOXA-AS3;HOXA-AS3;HOXA6;HOXA3;HOXA-AS3;HOXA-AS3 ///
HOXA6;HOXA-AS3;HOXA-AS3;HOXA6;HOXA3;HOXA-AS3;HOXA-AS3
</td>
<td style="text-align:left;width: 4cm; ">
epi_manova_3
</td>
<td style="text-align:left;width: 4cm; ">
ENSR00000325316
</td>
<td style="text-align:left;">
Promoter
</td>
</tr>
<tr>
<td style="text-align:left;width: 4cm; ">
MAN2B1
</td>
<td style="text-align:left;width: 4cm; ">
epi_manova_4
</td>
<td style="text-align:left;width: 4cm; ">
ENSR00001156259///ENSR00001156260
</td>
<td style="text-align:left;">
Promoter///CTCF Binding Site
</td>
</tr>
<tr>
<td style="text-align:left;width: 4cm; ">
AC093627.6;AC093627.6 /// AC093627.6;AC093627.6 ///
AC093627.6;AC093627.6 /// AC093627.6;AC093627.6 ///
AC093627.6;AC093627.6 /// AC093627.6;AC093627.6
</td>
<td style="text-align:left;width: 4cm; ">
epi_manova_5
</td>
<td style="text-align:left;width: 4cm; ">
ENSR00000324412
</td>
<td style="text-align:left;">
CTCF Binding Site
</td>
</tr>
</tbody>
</table>
</div>
<div class="section level3">
<h3 id="visualización-de-las-epimutaciones">Visualización de las epimutaciones<a class="anchor" aria-label="anchor" href="#visualizaci%C3%B3n-de-las-epimutaciones"></a>
</h3>
<p>La función <code><a href="https://rdrr.io/pkg/epimutacions/man/plot_epimutations.html" class="external-link">plot_epimutations()</a></code> dibuja una epimutación en
su contexto genómico, con los valores de metilación del caso en rojo,
los de los controles en líneas negras discontínuas y la media de la
población en azul:</p>
<div class="sourceCode" id="cb32"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="https://rdrr.io/pkg/epimutacions/man/plot_epimutations.html" class="external-link">plot_epimutations</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/pkg/BiocGenerics/man/as.data.frame.html" class="external-link">as.data.frame</a></span><span class="op">(</span><span class="va">epi_mvo</span><span class="op">[</span><span class="fl">15</span>,<span class="op">]</span><span class="op">)</span>, <span class="va">methy</span><span class="op">)</span></span></code></pre></div>
<p><img src="epimutacions_cast_files/figure-html/plot_default-1.png" width="700"></p>
<p>Si ponemos el argumento <code>gene_annot == TRUE</code>, el gráfico
incluye también las anotaciones génicas:</p>
<div class="sourceCode" id="cb33"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">p</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/epimutacions/man/plot_epimutations.html" class="external-link">plot_epimutations</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/pkg/BiocGenerics/man/as.data.frame.html" class="external-link">as.data.frame</a></span><span class="op">(</span><span class="va">epi_mvo</span><span class="op">[</span><span class="fl">1</span>,<span class="op">]</span><span class="op">)</span>, </span>
<span>                       methy <span class="op">=</span> <span class="va">methy</span>, </span>
<span>                       genes_annot <span class="op">=</span> <span class="cn">TRUE</span><span class="op">)</span></span></code></pre></div>
<div class="sourceCode" id="cb34"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="https://rdrr.io/r/graphics/plot.default.html" class="external-link">plot</a></span><span class="op">(</span><span class="va">p</span><span class="op">)</span></span></code></pre></div>
<p><img src="epimutacions_cast_files/figure-html/plot_genes_annot-1.png" width="700" style="display: block; margin: auto;"></p>
<p>Para graficar las marcas de cromatina H3K4me3, H3K27me3 y H3K27ac,
especifcamos el argumento <code>regulation = TRUE</code>:</p>
<ul>
<li>
<strong>H3K4me3</strong>: comunmente asociada con la activación de
la transcripción de genes cercanos.</li>
<li>
<strong>H3K27me3</strong>: usada en epigenética para detectar genes
inactivos.</li>
<li>
<strong>H3K27ac</strong>: asociada con una gran activación de la
transcripción y, por lo tanto, definida con una marca de enhancer
activador.</li>
</ul>
<div class="sourceCode" id="cb35"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">p</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/epimutacions/man/plot_epimutations.html" class="external-link">plot_epimutations</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/pkg/BiocGenerics/man/as.data.frame.html" class="external-link">as.data.frame</a></span><span class="op">(</span><span class="va">epi_mvo</span><span class="op">[</span><span class="fl">1</span>,<span class="op">]</span><span class="op">)</span>, </span>
<span>                       methy <span class="op">=</span>  <span class="va">methy</span>, </span>
<span>                       regulation <span class="op">=</span> <span class="cn">TRUE</span><span class="op">)</span></span></code></pre></div>
<div class="sourceCode" id="cb36"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="https://rdrr.io/r/graphics/plot.default.html" class="external-link">plot</a></span><span class="op">(</span><span class="va">p</span><span class="op">)</span></span></code></pre></div>
<p><img src="epimutacions_cast_files/figure-html/plot_genes_regulation-1.png" width="700" style="display: block; margin: auto;"></p>
</div>
</div>
<div class="section level2">
<h2 id="agradecimientos">Agradecimientos<a class="anchor" aria-label="anchor" href="#agradecimientos"></a>
</h2>
<p>Los autores quieren dar las gracias al equipo que colaboró en el
diseño inicial del paquete en el BioHackaton Europeo de 2020: Lordstrong
Akano, James Baye, Alejandro Caceres, Pavlo Hrab, Raquel Manzano y
Margherita Mutarelli. Los autores también quieren agradecer el apoyo de
la organización del BioHackaton Europeo de 2020.</p>
<p>Todos los miembros del equipo del <em>Project #5</em> que han
conribuido a este paquete:</p>
<table class="table">
<colgroup>
<col width="14%">
<col width="21%">
<col width="17%">
<col width="31%">
<col width="14%">
</colgroup>
<thead><tr class="header">
<th align="left">Name</th>
<th align="left">Surname</th>
<th align="center">ORCID</th>
<th align="left">Affiliation</th>
<th align="left">Team</th>
</tr></thead>
<tbody>
<tr class="odd">
<td align="left">Leire</td>
<td align="left">Abarrategui</td>
<td align="center">0000-0002-1175-038X</td>
<td align="left">Faculty of Medical Sciences, Newcastle University,
Newcastle-Upon-Tyne, UK; Autonomous University of Barcelona (UAB),
Barcelona, Spain</td>
<td align="left">Development</td>
</tr>
<tr class="even">
<td align="left">Lordstrong</td>
<td align="left">Akano</td>
<td align="center">0000-0002-1404-0295</td>
<td align="left">College of Medicine, University of Ibadan</td>
<td align="left">Development</td>
</tr>
<tr class="odd">
<td align="left">James</td>
<td align="left">Baye</td>
<td align="center">0000-0002-0078-3688</td>
<td align="left">Wellcome/MRC Cambridge Stem Cell Institute, University
of Cambridge, Cambridge CB2 0AW, UK; Department of Physics, University
of Cambridge, Cambridge CB2 3DY, UK</td>
<td align="left">Development</td>
</tr>
<tr class="even">
<td align="left">Alejandro</td>
<td align="left">Caceres</td>
<td align="center">-</td>
<td align="left">ISGlobal, Barcelona Institute for Global Health, Dr
Aiguader 88, 08003 Barcelona, Spain; Centro de Investigación Biomédica
en Red en Epidemiología y Salud Pública (CIBERESP), Madrid, Spain</td>
<td align="left">Development</td>
</tr>
<tr class="odd">
<td align="left">Carles</td>
<td align="left">Hernandez-Ferrer</td>
<td align="center">0000-0002-8029-7160</td>
<td align="left">Centro Nacional de Análisis Genómico (CNAG-CRG), Center
for Genomic, Regulation; Barcelona Institute of Science and Technology
(BIST), Barcelona, Catalonia, Spain</td>
<td align="left">Development</td>
</tr>
<tr class="even">
<td align="left">Pavlo</td>
<td align="left">Hrab</td>
<td align="center">0000-0002-0742-8478</td>
<td align="left">Department of Genetics and Biotechnology, Biology
faculty, Ivan Franko National University of Lviv</td>
<td align="left">Validation</td>
</tr>
<tr class="odd">
<td align="left">Raquel</td>
<td align="left">Manzano</td>
<td align="center">0000-0002-5124-8992</td>
<td align="left">Cancer Research UK Cambridge Institute; University of
Cambridge, Cambridge, United Kingdom</td>
<td align="left">Reporting</td>
</tr>
<tr class="even">
<td align="left">Margherita</td>
<td align="left">Mutarelli</td>
<td align="center">0000-0002-2168-5059</td>
<td align="left">Institute of Applied Sciences and Intelligent Systems
(ISASI-CNR)</td>
<td align="left">Validation</td>
</tr>
<tr class="odd">
<td align="left">Carlos</td>
<td align="left">Ruiz-Arenas</td>
<td align="center">0000-0002-6014-3498</td>
<td align="left">Centro de Investigación Biomédica en Red de
Enfermedades Raras (CIBERER), Barcelona, Spain; Universitat Pompeu Fabra
(UPF), Barcelona, Spain</td>
<td align="left">Reporting</td>
</tr>
</tbody>
</table>
</div>
<div class="section level2">
<h2 id="session-info">Session Info<a class="anchor" aria-label="anchor" href="#session-info"></a>
</h2>
<div class="sourceCode" id="cb37"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="https://rdrr.io/r/utils/sessionInfo.html" class="external-link">sessionInfo</a></span><span class="op">(</span><span class="op">)</span></span></code></pre></div>
<pre><code>R version 4.2.1 (2022-06-23 ucrt)
Platform: x86_64-w64-mingw32/x64 (64-bit)
Running under: Windows 10 x64 (build 19044)

Matrix products: default

locale:
[1] LC_COLLATE=Spanish_Spain.utf8  LC_CTYPE=Spanish_Spain.utf8   
[3] LC_MONETARY=Spanish_Spain.utf8 LC_NUMERIC=C                  
[5] LC_TIME=Spanish_Spain.utf8    

attached base packages:
[1] parallel  stats4    stats     graphics  grDevices utils     datasets 
[8] methods   base     

other attached packages:
 [1] rtracklayer_1.56.1                                 
 [2] IlluminaHumanMethylationEPICanno.ilm10b2.hg19_0.6.0
 [3] kableExtra_1.3.4                                   
 [4] minfi_1.42.0                                       
 [5] bumphunter_1.38.0                                  
 [6] locfit_1.5-9.6                                     
 [7] iterators_1.0.14                                   
 [8] foreach_1.5.2                                      
 [9] Biostrings_2.64.1                                  
[10] XVector_0.36.0                                     
[11] SummarizedExperiment_1.26.1                        
[12] Biobase_2.56.0                                     
[13] MatrixGenerics_1.8.1                               
[14] matrixStats_0.62.0                                 
[15] GenomicRanges_1.48.0                               
[16] GenomeInfoDb_1.32.4                                
[17] IRanges_2.30.1                                     
[18] S4Vectors_0.34.0                                   
[19] ExperimentHub_2.4.0                                
[20] AnnotationHub_3.4.0                                
[21] BiocFileCache_2.4.0                                
[22] dbplyr_2.2.1                                       
[23] BiocGenerics_0.42.0                                
[24] epimutacions_1.0.3                                 
[25] epimutacionsData_1.0.0                             
[26] BiocStyle_2.24.0                                   

loaded via a namespace (and not attached):
  [1] rappdirs_0.3.3                         
  [2] ragg_1.2.2                             
  [3] tidyr_1.2.1                            
  [4] ggplot2_3.3.6                          
  [5] bit64_4.0.5                            
  [6] knitr_1.40                             
  [7] DelayedArray_0.22.0                    
  [8] data.table_1.14.2                      
  [9] rpart_4.1.16                           
 [10] KEGGREST_1.36.3                        
 [11] RCurl_1.98-1.8                         
 [12] GEOquery_2.64.2                        
 [13] AnnotationFilter_1.20.0                
 [14] generics_0.1.3                         
 [15] GenomicFeatures_1.48.3                 
 [16] preprocessCore_1.58.0                  
 [17] RSQLite_2.2.16                         
 [18] bit_4.0.4                              
 [19] tzdb_0.3.0                             
 [20] webshot_0.5.3                          
 [21] xml2_1.3.3                             
 [22] httpuv_1.6.5                           
 [23] assertthat_0.2.1                       
 [24] xfun_0.32                              
 [25] hms_1.1.2                              
 [26] jquerylib_0.1.4                        
 [27] evaluate_0.16                          
 [28] promises_1.2.0.1                       
 [29] DEoptimR_1.0-11                        
 [30] fansi_1.0.3                            
 [31] restfulr_0.0.15                        
 [32] scrime_1.3.5                           
 [33] progress_1.2.2                         
 [34] DBI_1.1.3                              
 [35] htmlwidgets_1.5.4                      
 [36] reshape_0.8.9                          
 [37] purrr_0.3.4                            
 [38] ellipsis_0.3.2                         
 [39] dplyr_1.0.9                            
 [40] backports_1.4.1                        
 [41] bookdown_0.29                          
 [42] annotate_1.74.0                        
 [43] biomaRt_2.52.0                         
 [44] deldir_1.0-6                           
 [45] sparseMatrixStats_1.8.0                
 [46] vctrs_0.4.1                            
 [47] ensembldb_2.20.2                       
 [48] cachem_1.0.6                           
 [49] withr_2.5.0                            
 [50] Gviz_1.40.1                            
 [51] BSgenome_1.64.0                        
 [52] robustbase_0.95-0                      
 [53] checkmate_2.1.0                        
 [54] GenomicAlignments_1.32.1               
 [55] prettyunits_1.1.1                      
 [56] mclust_5.4.10                          
 [57] svglite_2.1.0                          
 [58] cluster_2.1.3                          
 [59] lazyeval_0.2.2                         
 [60] crayon_1.5.2                           
 [61] genefilter_1.78.0                      
 [62] pkgconfig_2.0.3                        
 [63] labeling_0.4.2                         
 [64] nlme_3.1-157                           
 [65] ProtGenerics_1.28.0                    
 [66] nnet_7.3-17                            
 [67] rlang_1.0.6                            
 [68] lifecycle_1.0.3                        
 [69] filelock_1.0.2                         
 [70] dichromat_2.0-0.1                      
 [71] rprojroot_2.0.3                        
 [72] graph_1.74.0                           
 [73] rngtools_1.5.2                         
 [74] base64_2.0.1                           
 [75] Matrix_1.5-1                           
 [76] Rhdf5lib_1.18.2                        
 [77] base64enc_0.1-3                        
 [78] png_0.1-7                              
 [79] viridisLite_0.4.1                      
 [80] rjson_0.2.21                           
 [81] bitops_1.0-7                           
 [82] rhdf5filters_1.8.0                     
 [83] blob_1.2.3                             
 [84] DelayedMatrixStats_1.18.1              
 [85] doRNG_1.8.2                            
 [86] stringr_1.4.1                          
 [87] nor1mix_1.3-0                          
 [88] readr_2.1.2                            
 [89] jpeg_0.1-9                             
 [90] scales_1.2.1                           
 [91] memoise_2.0.1                          
 [92] magrittr_2.0.3                         
 [93] plyr_1.8.7                             
 [94] zlibbioc_1.42.0                        
 [95] compiler_4.2.1                         
 [96] BiocIO_1.6.0                           
 [97] RColorBrewer_1.1-3                     
 [98] illuminaio_0.38.0                      
 [99] Rsamtools_2.12.0                       
[100] cli_3.3.0                              
[101] htmlTable_2.4.1                        
[102] Formula_1.2-4                          
[103] MASS_7.3-57                            
[104] tidyselect_1.2.0                       
[105] stringi_1.7.8                          
[106] textshaping_0.3.6                      
[107] highr_0.9                              
[108] yaml_2.3.5                             
[109] askpass_1.1                            
[110] latticeExtra_0.6-30                    
[111] ggrepel_0.9.1                          
[112] grid_4.2.1                             
[113] sass_0.4.1                             
[114] VariantAnnotation_1.42.1               
[115] tools_4.2.1                            
[116] rstudioapi_0.14                        
[117] foreign_0.8-82                         
[118] TxDb.Hsapiens.UCSC.hg19.knownGene_3.2.2
[119] gridExtra_2.3                          
[120] farver_2.1.1                           
[121] digest_0.6.29                          
[122] BiocManager_1.30.18                    
[123] shiny_1.7.2                            
[124] quadprog_1.5-8                         
[125] Rcpp_1.0.9                             
[126] siggenes_1.70.0                        
[127] BiocVersion_3.15.2                     
[128] later_1.3.0                            
[129] OrganismDbi_1.38.1                     
[130] org.Hs.eg.db_3.15.0                    
[131] httr_1.4.4                             
[132] AnnotationDbi_1.58.0                   
[133] biovizBase_1.44.0                      
[134] colorspace_2.0-3                       
[135] rvest_1.0.3                            
[136] XML_3.99-0.10                          
[137] fs_1.5.2                               
[138] splines_4.2.1                          
[139] RBGL_1.72.0                            
[140] pkgdown_2.0.6                          
[141] multtest_2.52.0                        
[142] systemfonts_1.0.4                      
[143] xtable_1.8-4                           
[144] jsonlite_1.8.0                         
[145] R6_2.5.1                               
[146] Hmisc_4.7-1                            
[147] pillar_1.8.1                           
[148] htmltools_0.5.3                        
[149] mime_0.12                              
[150] glue_1.6.2                             
[151] fastmap_1.1.0                          
[152] BiocParallel_1.30.3                    
[153] interactiveDisplayBase_1.34.0          
[154] beanplot_1.3.1                         
[155] codetools_0.2-18                       
[156] Homo.sapiens_1.3.1                     
[157] utf8_1.2.2                             
[158] isotree_0.5.17                         
[159] lattice_0.20-45                        
[160] bslib_0.4.0                            
[161] tibble_3.1.7                           
[162] curl_4.3.2                             
[163] GO.db_3.15.0                           
[164] openssl_2.0.2                          
[165] interp_1.1-3                           
[166] survival_3.3-1                         
[167] limma_3.52.3                           
[168] rmarkdown_2.16                         
[169] desc_1.4.1                             
[170] munsell_0.5.0                          
[171] rhdf5_2.40.0                           
[172] GenomeInfoDbData_1.2.8                 
[173] HDF5Array_1.24.2                       
[174] reshape2_1.4.4                         
[175] gtable_0.3.1                           </code></pre>
</div>
<div class="section level2">
<h2 class="unnumbered" id="referencias">Referencias<a class="anchor" aria-label="anchor" href="#referencias"></a>
</h2>
<div id="refs" class="references csl-bib-body hanging-indent">
<div id="ref-aref2019" class="csl-entry">
Aref-Eshghi, Erfan, Eric G. Bend, Samantha Colaiacovo, Michelle Caudle,
Rana Chakrabarti, Melanie Napier, Lauren Brick, et al. 2019.
<span>“Diagnostic Utility of Genome-Wide DNA Methylation Testing in
Genetically Unsolved Individuals with Suspected Hereditary
Conditions.”</span> <em>The American Journal of Human Genetics</em>.
https://doi.org/<a href="https://doi.org/10.1016/j.ajhg.2019.03.008" class="external-link">https://doi.org/10.1016/j.ajhg.2019.03.008</a>.
</div>
<div id="ref-barbosa2018identification" class="csl-entry">
Barbosa, Mafalda, Ricky S Joshi, Paras Garg, Alejandro Martin-Trujillo,
Nihir Patel, Bharati Jadhav, Corey T Watson, et al. 2018.
<span>“Identification of Rare de Novo Epigenetic Variations in
Congenital Disorders.”</span> <em>Nature Communications</em> 9 (1):
1–11.
</div>
<div id="ref-jaffe2012bump" class="csl-entry">
Jaffe, Andrew E, Peter Murakami, Hwajin Lee, Jeffrey T Leek, M Daniele
Fallin, Andrew P Feinberg, and Rafael A Irizarry. 2012. <span>“Bump
Hunting to Identify Differentially Methylated Regions in Epigenetic
Epidemiology Studies.”</span> <em>International Journal of
Epidemiology</em> 41 (1): 200–209.
</div>
</div>
</div>
  </main><aside class="col-md-3"><nav id="toc"><h2>On this page</h2>
    </nav></aside>
</div>



    <footer><div class="pkgdown-footer-left">
  <p></p>
<p>Developed by Natàlia Carreras-Gallo, Laura Balagué-Dobón, Xavier Escribà-Montagut, Juan R Gonzàlez.</p>
</div>

<div class="pkgdown-footer-right">
  <p></p>
<p>Site built with <a href="https://pkgdown.r-lib.org/" class="external-link">pkgdown</a> 2.0.6.</p>
</div>

    </footer>
</div>

  

  

  </body>
</html>
